### PR TITLE
add missing test for base==<unknown> case

### PR DESCRIPTION
**Pull request by tveronezi**

### CODE REVIEW

File: core/module_specifier.rs [hunk 1]
```diff
@@ -66,11 +66,20 @@ impl ModuleSpecifier {
     self.0.as_str()
   }
 
+  pub fn resolve_import(
```
Comment (tveronezi) [inline core/module_specifier.rs:69]: reason: making it possible to manually set the curr_dir without changing every place where the original resolve import function is used.

**Comment by bartlomieju** [comment]
@tveronezi does this PR solve specific problem?

**Comment by tveronezi** [comment]
@bartlomieju maybe. The meat of it is the addition of the missing test.

The original code behaves like this with the new tests:

```
---- module_specifier::tests::test_resolve_import stdout ----
thread 'module_specifier::tests::test_resolve_import' panicked at 'assertion failed: `(left == right)`
  left: `"file:///awesome.ts"`,
 right: `"file:///home/tveronezi/Documents/source/oss/deno/core/awesome.ts"`', core/module_specifier.rs:298:7
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The original code joins the base ("which is the unkown string") to the current directory....

```
let path = current_dir().unwrap().join(base);
let base = Url::from_file_path(path).unwrap()
base.join(&specifier).map_err(InvalidUrl)?
```
So, for cases where curr dir is (for example) "/opt/app" and the specifier is "/awesome.ts", the final path is "file:///awesome.ts". 

This PR makes is in a way that the same input generates "file:///opt/app/awesome.ts". 

This PR brings up one question: Is the original behavior the expected behavior?

If that's an `yup` then the PR is good to go; If that's a `nop`, I will update the tests to match the original behavior.

**Comment by bartlomieju** [comment]
> So, for cases where curr dir is (for example) "/opt/app" and the specifier is "/awesome.ts", the final path is "file:///awesome.ts".

That's correct; `/awesome.ts` is an absolute path so it should resolve to `file:///awesome.ts`. 

The `<unknown>` is not a valid base for URL so it should probably error with (RelativeUrlWithCannotBeABaseBase)[`https://docs.rs/url/2.1.1/url/enum.ParseError.html#variant.RelativeUrlWithCannotBeABaseBase`]

**Comment by tveronezi** [comment]
I will create a new PR with just the test changes.

**Pull request closed without merging**